### PR TITLE
refactor(accent): make language detection publish-only

### PR DESCRIPTION
**Pull request by barad1tos**
── Summary ─────────────────────────────────

C7 from the architecture review (its "speculative / higher risk" candidate — landed with the minimal restructuring). A cold-cache resolve scheduled a background scan whose completion **re-entered the resolve+apply pipeline from inside `ProjectLanguageDetector`** — the reason disposal-race defenses dominated its 607 lines. The detector now only publishes the existing `scanCompleted` Topic; one project-scoped subscriber (`ScanCompletionAccentRefresher`) owns the re-resolve+apply reaction with the verbatim body, gates, and log messages.

── Changes ─────────────────────────────────

| Type | Where | What |
|------|-------|------|
| Refactor | `src/main/kotlin/dev/ayuislands/accent/ScanCompletionAccentRefresher.kt:1` | Project-scoped light service; subscribes in `init` with its own Disposable as parent (Pattern E); outcome gate mirrors the deleted verdict gate exactly (Detected/Polyglot → refresh, Unavailable → skip) |
| Refactor | `src/main/kotlin/dev/ayuislands/accent/ProjectLanguageDetector.kt` | −84 lines: scan body is detectAndCache → classify → publish only; `tryRefreshAccentAfterCacheableScan`/`refreshAccentOnEdt` deleted (KDoc reasoning relocated with the code) |
| Refactor | `src/main/kotlin/dev/ayuislands/AyuIslandsStartupActivity.kt` | Refresher installs as the FIRST project-scoped statement — before the cold-cache resolve that schedules the first scan (closes a subscriber-not-yet-installed race the baked-in version never had), and before the external-theme early return (mid-session theme-switch behavior preserved 1:1); locked by a source-order test |
| Test | `src/test/kotlin/dev/ayuislands/accent/ScanCompletionAccentRefresherTest.kt:1` | 10 tests, all driven through the real `init` subscription; +3 new locks that fell out: rejected-hex swap suppression (previously untested), Unavailable gate, disposable anchoring |

── Validation ──────────────────────────────

```bash
./gradlew test detekt ktlintCheck koverVerify
```
Expected: green, 2928 tests; refresher 100% line/branch, detector 95.7%.

Manual: open a polyglot project cold (no forced language, language override configured) — after indexing settles, the accent must flip to the detected language's override without touching Settings; `grep "Post-scan accent refresh" idea.log` shows the subscriber path.

── Notes ───────────────────────────────────

- One deliberate, strictly-safer delta: a disposal-dropped publish now means no refresh for a dying project (previously the refresh could still run). No test pinned the old apply-before-publish order — verified before reworking the two tests that pinned the internal two-EDT-hop structure (wiring-only changes, every behavioral assertion intact).
- Declarative `plugin.xml` `projectListeners` was considered and rejected: no precedent in this plugin and more platform-API risk on the 2025.1 pin than a light service with an init-block subscription.

## Summary by Sourcery

Refactor language detection so ProjectLanguageDetector becomes a publish-only event source and move the post-scan accent refresh logic into a dedicated project service subscribed to scanCompleted, with tests and documentation updated to lock in the new one-way flow and startup ordering.

Enhancements:
- Decouple accent refresh from language detection by introducing a project-scoped ScanCompletionAccentRefresher service subscribed to scanCompleted events.
- Restrict ProjectLanguageDetector to publishing scan outcomes without re-entering the accent resolve+apply pipeline.
- Ensure startup installs the scan-completion accent refresher before the first accent resolve to avoid missing the initial background scan.
- Strengthen the ProjectLanguageDetectionListener contract to document the new accent refresher subscriber and one-way event flow.

Documentation:
- Update feature verification metadata in docs/features.yml to reflect the new accent-detection behavior and verified commit.

Tests:
- Move and expand accent refresh behavior tests into ScanCompletionAccentRefresherTest, exercising subscription wiring and edge cases end-to-end.
- Adapt ProjectLanguageDetectorTest and AyuIslandsStartupActivityTest to the publish-only detector model, including source-order guards for refresher installation.

### CODE REVIEW

File: src/main/kotlin/dev/ayuislands/AyuIslandsStartupActivity.kt [hunk 1]
```diff
@@ -39,6 +40,18 @@ internal class AyuIslandsStartupActivity : ProjectActivity {
...
+        // Language detection is a one-way event: the detector only publishes
+        // scanCompleted, and this project service is the single subscriber that
+        // owns the post-scan re-resolve + accent apply. Instantiate it BEFORE
+        // the first accent resolve below — runStartupAccentOnEdt can schedule
+        // the first background scan (EDT resolve on a cold cache), and the
+        // scan's completion must not fire into an empty subscriber list.
+        // Installed before the external-theme early return on purpose: the
+        // refresher self-gates on AyuVariant.detect(), mirroring the reaction
+        // that previously lived unconditionally inside the detector.
+        ScanCompletionAccentRefresher.getInstance(project)
```
[the remark this answers is not in the document]
Comment (barad1tos) [reply src/main/kotlin/dev/ayuislands/AyuIslandsStartupActivity.kt:53]: Analyzed — the flow converges in both orderings, because project B's own startup activity installs B's refresher FIRST and only then applies: (1) B's scan (scheduled by A's focused-resolve) completes BEFORE B's activity → publish has no subscriber, but B's activity then runs install→resolve against the now-WARM cache and applies the correct override immediately; (2) B's scan completes AFTER B's activity started → B's refresher is already subscribed and fires normally. The only visible artifact is a transient fallback accent for B during multi-project startup, bounded by B's own activity — documented in RescanLanguageAction's triage KDoc. Installing B's service from A's activity would work too, but adds cross-project service instantiation for a window that self-heals.
